modules/eido/validate/main.nf

```diff
@@ -0,0 +1,37 @@
+process EIDO_VALIDATE {
+    tag '$samplesheet'
+    label 'process_low'
+
+    conda (params.enable_conda ? "conda-forge::eido=0.1.8" : null)
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://containers.biocontainers.pro/s3/SingImgsRepo/eido/0.1.9_cv1/eido_0.1.9_cv1.sif' :
+        'biocontainers/eido:0.1.9_cv1' }"
+
+    input:
+    path samplesheet
+    path schema
+
+    output:
+    path "versions.yml"           , emit: versions
+    path "conversion.out"         , emit: conversion
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    """
+    if [[ $samplesheet = "https://" || $samplesheet = "http://" ]]; then
+        SAMPLESHEET_PATH=$samplesheet
+    else
+        SAMPLESHEET_PATH=\$(readlink -f $samplesheet)
+    fi
```

Comment:
```suggestion
    if [[ $samplesheet = "https://" || $samplesheet = "http://" ]]; then
        SAMPLESHEET_PATH=$samplesheet
    else
        SAMPLESHEET_PATH=\$(readlink -f $samplesheet)
    fi
```

Is this necessary?

Nextflow's `path` directive (in the input channel definition) auto-detects whether a file is 'remote' or not already. If it is, it downloads it for you and symlinks into the working directory of the process. Same goes for local files (just skips the download). You shouldn't need to do readlink step. 

You should be able to pass `$samplesheet` directly to the `validate` command itself I think